src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -472,6 +472,9 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 				await this._connectionManagementService.deleteConnection(profile);
 			}
 			const connectionProfile = this.getConnectionInTreeInput(profile.id);
+			if (!connectionProfile.savePassword) {
```

Comment:
What was the bug you were seeing with this?

Reply:
ads was not asking for the password the next time the user connected with this connection. Since the password was already saved in the connection from last session.

Reply:
This is only an issue in the async tree I'm guessing then, right? Because the other tree makes new inputs when things change?

Reply:
Yup, an async tree only issue since we do not get rid of connections objects when they are disconnected.

Reply:
Could you add that as a comment here so we aren't confused later on why this is happening?